### PR TITLE
Add Autocomplete for Cities [#6]

### LINKED ISSUE #6: Autocomplete for cities
At the moment, when you type in a city, it won't help you finish your entry with suggestions. A cool feature would be if it had a drop down menu that suggested locations similar to the name. For example, if I type in `Cambridge`, it could tell me `Cambridge, MA, USA` or `Cambridge, UK` to help users get the right answers!

### PR DESCRIPTION
**Pull request by ditsuke**
## Context
Adds autocomplete for city names. We decided against using the Google Maps API (required billing), instead going with a city list @yyjlincoln found and doing the matching client side. The suggestions are pretty good from our tests.

## Testing
Since the Netflify preview is not working too well, we'll need to test this locally:
- Clone this PR locally (using [Github CLI](https://cli.github.com/): `gh pr checkout 20`)
- `yarn install`
- `yarn start`
- Play around with the search box.
- Switch back to the branch you were working with: `git switch -`

## Authors
@ditsuke @yyjlincoln 

- [x]  My Pod Leader knows I'm working on this Pull Request
- [x] I've explained what the Pull Request is adding.
- [x] I've explained why this is important.

Closes #6

### CODE REVIEW

**Comment by harshsharma6401** [comment]
The auto-completion functionality is working fine.  You can display that in a dropdown list like general search bars. Like when we search at google. Maybe you can add a div block for dropdown list after this line <input type="text" value={city} onChange={event => setCity(event.target.value)} />.

**Comment by ditsuke** [comment]
> The auto-completion functionality is working fine. You can display that in a dropdown list like general search bars. Like when we search at google. Maybe you can add a div block for dropdown list after this line <input type="text" value={city} onChange={event => setCity(event.target.value)} />.

We're yet to work on it, but we'll add that functionality soon before graduating this PR from a draft.

**Comment by ditsuke** [comment]
Ready for review 

cc @harshsharma6401 @KhudaDad414 @CIPHERTron @KshitijDarekar @psahithireddy

**Review by yyjlincoln — APPROVED** [review]
Looks great! We should probably add a note to tell other contributors to do `yarn` again after pulling the latest changes - as we added a new dependency

**Comment by yyjlincoln** [comment]
Requesting review: @gmelodie @ditsuke @Dhruv-Sachdev1313 @Aniket762

**Comment by harshsharma6401** [comment]
It's much better. Getting a similar experience as we get on other websites. Its good to go.

**Review by harshsharma6401 — APPROVED** [review]

**Review by psahithireddy — APPROVED** [review]
works well.We can go ahead and merge.

**Comment by psahithireddy** [comment]
> Looks great! We should probably add a note to tell other contributors to do `yarn` again after pulling the latest changes - as we added a new dependency

agreed.

**Comment by ditsuke** [comment]
Updated with testing instructions!

**Comment by KshitijDarekar** [comment]
LGTM. Also, the testing instructions helped.

**Review by CIPHERTron — APPROVED** [review]
LGTM 🎉

File: src/App.js [hunk 1]
```diff
@@ -23,9 +36,9 @@ function App() {
         (error) => {
           setIsLoaded(true);
           setError(error);
-        }
-      )
-  }, [city])
+        },
+      );
+  }, [city]);
```
Comment (ditsuke) [inline src/App.js:41]:
```suggestion
        }
      )
  }, [city])
```

**Comment by ditsuke** [comment]
Force pushed to revert a code-style change. No material changes.

**Comment by ditsuke** [comment]
Marking good to merge/ RTC.

**Pull request merged**